core/src/main/java/org/infinispan/conflict/EntryMergePolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.conflict;
+
+import java.util.List;
+
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.partitionhandling.PartitionHandling;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.1
+ */
+public interface EntryMergePolicy<K, V> {
+
+   /**
+    * This method is called by {@link ConflictManager#resolveConflicts()} for each conflict discovered to determine
+    * which {@link CacheEntry} should be utilised. This merge policy is used when a user explicitly calls {@link ConflictManager#resolveConflicts()}
+    * as well as when a partition merge occurs with {@link PartitionHandling#ALLOW_READ_WRITES} set.
+    *
+    * In the event of a partition merge, we define the preferred partition as the partition whom's coordinator is coordinating
+    * the current merge.
```

Comment:
When you're resolving conflicts on demand, what is the preferred partition?
When you're merging a cluster with two degraded partitions, but one of the partitions contains all owners of given entry and the second one has the future coordinator, does it mean that you'll wipe out the data? (with PREFERRED_ALWAYS policy)